src/Core/Checkout/Cart/SalesChannel/CartOrderRoute.php

```diff
@@ -69,19 +74,15 @@ public function order(Cart $cart, SalesChannelContext $context, RequestDataBag $
         // we use this state in stock updater class, to prevent duplicate available stock updates
         $context->addState('checkout-order-route');
 
-        $calculatedCart = $this->cartCalculator->calculate($cart, $context);
-
-        $response = $this->checkoutGatewayRoute->load(new Request($data->all(), $data->all()), $cart, $context);
-        $calculatedCart->addErrors(...$response->getErrors());
-
-        $this->taxProviderProcessor->process($calculatedCart, $context);
+        $placed = $this->extensions->publish(
+            name: CheckoutPlaceOrderExtension::NAME,
+            extension: new CheckoutPlaceOrderExtension($cart, $context, $data),
+            function: $this->place(...)
```

Comment:
Just a minor remark:

If you pass arguments via the destructuring operator (...) directly to $this->place(...), the place method becomes dependent on the exact structure and order of the parameters being provided by the route.

This can lead to mistakes in the future, please provide the params directly

Reply:
I tried out your suggestion but this is not working. This is also not a call, it is a reference to the `this->place` method. The function will be called in the `\Shopware\Core\Framework\Extensions\ExtensionDispatcher::publish`

![image](https://github.com/user-attachments/assets/49a10fef-f425-45df-87f8-40bd561fd9a3)


Which makes it required, to have exact the same parameter names and order. 🙈